### ISSUE TITLE
auk_zerofill error: long vectors not supported yet

### ISSUE THREAD

**Issue by tdlan80** [CornellLabofOrnithology/auk#42]
I wanted to use `auk::auk_zerofill` function to convert a presence only species matrix (produced via `auk_ebd`) into a presence-absence dataset. The presence-only file has been already filtered by `auk_filter` to extract only the data I need for my analyses. When I ran `auk::auk_zerofill`, the arguments `collapse `, `unique`, `rollup `, and `drop_higher` were `=T`. I get the following error message:

> Error in vec_slice(x, idx) : 
  long vectors not supported yet: ../include/Rinlinedfuns.h:535

Does this mean the my number of columns, i.e., the number of species is too high? Should I subset (may be via `auk_split` or `auk_select`) my presence only matrix, run  `auk::auk_zerofill` for each subset and then run `collapse_zerofill()`? But, not sure if that does not helps at all?

I do need to keep all my species at the end since I am looking at community wide analyses. Any suggestions on understanding this error message and troubleshooting it will be appreciated.

**Comment by mstrimas** [comment]
I've never seen this error. How many rows are in your data frame?

**Comment by tdlan80** [comment]
for the actual dataset, after cleaning it up, I have 110790 rows. for the sampling events data set, I have 7868598 rows.

**Comment by mstrimas** [comment]
my guess is that the large number of rows in your sampling event data, combined with a large number of species, is resulting in a matrix that's too big for R to handle (e.g. [see this](https://stackoverflow.com/questions/24335692/large-matrices-in-r-long-vectors-not-supported-yet)). Probably the easiest thing is to process the data for each species independently. You could use `auk_split()` to divide the EBD up into species files and zero-fill each one rather than trying to do all species at once.

**Comment by tdlan80** [comment]
@mstrimas Thanks for the suggestion. I did read threads similar to the one you linked. I was too, planning on sub-setting my dataframe. So, this issue relates to limitation of R, not that of auk? And, nothing to so with memory?

**Comment by mstrimas** [comment]
Without having the data in from of me, it's not clear to me what's going on exactly. The error makes it seem like it's not a memory issue, rather a limitation in R with maximum size of matrices. I suspect I probably haven't written `auk_zerofill()` in the most efficient way, so it may be possible to do the zero filling without causing this error. If you're able to pinpoint exactly which line `auk_zerofill()` is causing the error that might help, e.g. you could try calling `traceback()` after the error is raised to see where it's coming from.

**Comment by tdlan80** [comment]
Let me try that and post back the outcome. is there a way to send the data your way? OR should I describe you the spatial extent of the data I extracted fro EBD?

**Comment by tdlan80** [comment]
I think I figured out the issue I have had. when did `auk_filter`, I had my EBD file data was only for a single county with 5 years data-- I downloaded that data for that specific county. During filtering, although I created filters via `auk_ebd` for both sampling event data and EBD data, I did not downsize the sampling event data to the same county as my EBD data. This resulted in an unnecessarily large sampling event data file at the `auk_zerofill`. I re-created `auk_ebd` filters and included the specific county via `auk_county`, and ran that filter on both EBD and sampling events. Now, `auk_zerofill` works fine. I did not have to use `auk_split()`.
